code/datums/components/full_auto_fire.dm

```diff
@@ -148,6 +148,8 @@
 		modifiers["icon-x"] = num2text(ABS_PIXEL_TO_REL(text2num(modifiers["icon-x"])))
 		modifiers["icon-y"] = num2text(ABS_PIXEL_TO_REL(text2num(modifiers["icon-y"])))
 		params = list2params(modifiers)
+		if(source.pixel_x != 0 || source.pixel_y != 0)
+			target = darkness_adjust(target, shooter)
```

Comment:
This issue should be happening only with regular clicks, not with autofire, the solution for it being in the code right here over your addition.
Is that not correct?

Reply:
it does work with no tile offset, but doesnt work if the client is zoomed